[BUG] Incorrect CMake Minimum Version Causes Build Failure When BUILD_DEPENDENCIES=off

**Describe the bug**
Building the library without dependencies (using definition `-DBUILD_DEPENDENCIES=off`) with CMake at version < 3.12 will cause CMake errors that look similar to the following, even when libcurl is available on the system.

```
CMake Error at CMakeLists.txt:277 (add_library):
  Target "cproducer" links to target "CURL::libcurl" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
```

This seems to be because of the way in which imported targets are specified in earlier versions of CMake. Notice in the CMake documentation that the `FindCURL` module does not define the `CURL::libcurl` target in versions earlier than 3.12.4 ([here](https://cmake.org/cmake/help/v3.11/module/FindCURL.html) is the documentation for the module at version 3.11).

**To Reproduce**
Steps to reproduce the behavior:
1. Downgrade CMake to a version below 3.12.
2. Follow the build instructions outlined in the README, but make sure to run `cmake .. -DBUILD_DEPENDENCIES=off`.
3. Notice errors.

**Expected behavior**
CMake should proceed without error when using minimum version specified in the CMakeLists.txt

**SDK version number**
master commit 665568fcc23c1f2da720b6f06780acb7f59d25a1

**Open source building**
libcurl/7.58.0

**Platform (please complete the following information):**
 - OS: Linux
 - Processor: x86_64
 - Version 18.04

--- a/CMake/Dependencies/libkvscproducer-CMakeLists.txt
+++ b/CMake/Dependencies/libkvscproducer-CMakeLists.txt
@@ -7,7 +7,7 @@ include(ExternalProject)
 # clone repo only
 ExternalProject_Add(libkvscproducer-download
 	GIT_REPOSITORY    https://github.com/awslabs/amazon-kinesis-video-streams-producer-c.git
-    GIT_TAG           81305d2796b5d571a8c19edc59c0463975f97c8c
+    GIT_TAG           ca3401496223bd536faff4eaa8f14eb01c0642f9
     SOURCE_DIR        "${CMAKE_CURRENT_BINARY_DIR}/kvscproducer-src"
 	BINARY_DIR        "${CMAKE_CURRENT_BINARY_DIR}/kvscproducer-build"
 	CONFIGURE_COMMAND ""

